1-models/articles-model.js

```diff
@@ -31,3 +31,22 @@ exports.selectArticleById = (article_id) => {
     }
   });
 };
+
+exports.updateArticleVotes = (article_id, inc_votes) => {
+  return this.selectArticleById(article_id)
+    .then((article) => {
+    if (article.votes + inc_votes < 0) {
+      return Promise.reject({
+        status: 400,
+        message: "ERROR ! resulting votes cannot be negative",
+      });
+    } else {
+      const queryString = `UPDATE articles SET votes = votes + $1 WHERE article_id = $2`;
```

Comment:
remember that by adding a RETURNING *; to the end of this, it will return the updated article object. This means you won't need that SELECT, and the resulting code will be more streamlined!